Add documentation for virtctl create command (#705)
kubevirt/user-guide: docs/virtual_machines/creating_vms.md

Reviewer: Can you add examples / explanation for creating cluster-wide or namespaced resources?

virtctl create vm -h
```

> # Creating InstanceTypes

Reviewer 2: `Instance Types`

> The virtctl subcommand `create instancetype` allows easy creation of an InstanceType

Reviewer 2: instance type

manifest from the command line. This manifest can be later used to create your VirtualMachine.
The command also provides several flags that can be used to create your desired manifest.

> There are two required flags that need to be specified: the number of CPUs and the amount of

Reviewer 2: vCPUs

memory to be used. Additionally, there are several optional flags that can be used, such as specifying
a list of GPUs for passthrough, choosing the desired IOThreadsPolicy, or simply providing
the name of our InstanceType.

> 

Reviewer 2: Can you provide a simple example with the required fields?

For a complete list of flags and their descriptions, use the following command:

```shell
virtctl create instancetype -h
```

# Creating Preferences

> The virtctl subcommand `create preference` allows easy creation of a Preference

Reviewer 2: preference

virtctl create vm -h
```

# Creating Instance Types

The virtctl subcommand `create instancetype` allows easy creation of an instance type
manifest from the command line. The command also provides several flags that can be used
to create your desired manifest.

There are two required flags that need to be specified: the number of vCPUs and the amount of

> memory to be requested. Additionally, there are several optional flags that can be used, such as specifying

Author: The purpose of this paragraph was to provide an overview of what can be done with this command. To view all the available flags, there's a hint below to use `virctl create instancetype -h`, which will list all the options. I was trying to stick with already present docs style that is here for [`create vm`](https://kubevirt.io/user-guide/virtual_machines/creating_vms/#creating-virtualmachines). But if others find it also useful we could create a list of available flags for each command.

a list of GPUs for passthrough, choosing the desired IOThreadsPolicy, or simply providing
the name of our InstanceType.

By default, the command creates the cluster-wide resource. If the user wants to create the namespaced version,
they need to provide the namespaced flag. The namespace name can be specified by using the namespace flag.

For a complete list of flags and their descriptions, use the following command:

```shell
virtctl create instancetype -h
```

# Creating Preferences

The virtctl subcommand `create preference` allows easy creation of a preference
manifest from the command line. This command serves as a starting point to create
the basic structure of a manifest, as it does not allow specifying all of the options that

> are supported in preferences.

Author: This is about the `create preference` command. While it might not be clear from the PR preview, on the [documentation website](https://kubevirt.io/user-guide/virtual_machines/creating_vms/), it is in my opinion structured quite well into blocks with headers etc.

The current set of flags allows us, for example, to specify the preferred CPU topology, machine type
or a storage class.

By default, the command creates the cluster-wide resource. If the user wants to create the namespaced version,

> they need to provide the namespaced flag. The namespace name can be specified by using the namespace flag.

Reviewer 3: ```suggestion
they need to provide the namespaced flag `--namespace  <desired namepace>`.
```

For a complete list of flags and their descriptions, use the following command:

```shell

> virtctl create instancetype -h

Reviewer 4: preferences have the same help command as instance types?

virtctl create vm -h
```

# Creating Instance Types

The virtctl subcommand `create instancetype` allows easy creation of an Instance type

> manifest from the command line. This manifest can be later used to create your VirtualMachine.

Reviewer: How can the manifest be used to create a VirtualMachine? What do you mean by that?

Author: I will drop this line

The command also provides several flags that can be used to create your desired manifest.

There are two required flags that need to be specified: the number of vCPUs and the amount of

> memory to be used. Additionally, there are several optional flags that can be used, such as specifying

Reviewer: s/used/requested/?

a list of GPUs for passthrough, choosing the desired IOThreadsPolicy, or simply providing
the name of our InstanceType.

```shell
virtctl create instancetype --cpu 2 --memory 256Mi
```

For a complete list of flags and their descriptions, use the following command:

```shell
virtctl create instancetype -h
```

# Creating Preferences

The virtctl subcommand `create preference` allows easy creation of a preference
manifest from the command line. This command serves as a starting point to create

> the basic structure of a manifest, as it does not allow specifying all of the options that
> can be manually specified.

Reviewer: ```suggestion
the basic structure of a manifest, as it does not allow specifying all of the options that
are supported in preferences.
```

virtctl create vm -h
```

> # Creating Instance Types

Reviewer 5: Would you mind moving this to its own page or make it a part of the "Instance types and preferences" page? My concern is that this "Creating VirtualMachines" is one of the first introductory pages a new user would go through. Mixing an advanced topic such as instance types into it may make it overwhelming.

Reviewer 5: You can leave here a note that "The flow can be simplified by using instance types" and reference the relevant page

Reviewer 2: I strongly disagree with this, instance types aren't an advance feature and if anything we want to drive more initial users towards using them with `virtctl create vm` as anything more involved is likely better done in a full blown manifest.

Reviewer 2: Apologies, I see what you mean in terms of ordering however, having this later after the initial `Creating VirtualMachines on a cluster`  section might make more sense.

Reviewer 5: I see you point in advertising this as "the way to manage VMs". I just don't want to scare people away before they manage to create their first VM. +1 on moving the instance types section lower in the page, I would be happy to approve this then

Author: I have moved it below the `Creating VirtualMachines on a cluster`

Reviewer 5: @opokornyy do you also want to update the link in https://github.com/kubevirt/user-guide/pull/705/files#diff-ba5436ef866e9da76ce83c188f741af99579d076a7f4a17247cbb448965cd032R176 to point directly to the new place?

Author: I would keep the link as it is, it should point to the usage of virtctl to create instance types, preferences, and virtual machines.

The virtctl subcommand `create instancetype` allows easy creation of an instance type
manifest from the command line. The command also provides several flags that can be used
to create your desired manifest.

There are two required flags that need to be specified: the number of vCPUs and the amount of
memory to be requested. Additionally, there are several optional flags that can be used, such as specifying
a list of GPUs for passthrough, choosing the desired IOThreadsPolicy, or simply providing
the name of our instance type.

By default, the command creates the cluster-wide resource. If the user wants to create the namespaced version,
they need to provide the namespaced flag. The namespace name can be specified by using the namespace flag.

For a complete list of flags and their descriptions, use the following command:

```shell
virtctl create instancetype -h
```

## Examples

> Create a manifest for a VirtualMachineClusterInstancetype with a required flags

Reviewer 2: nit - `the required --cpu and --memory flags`

```shell
virtctl create instancetype --cpu 2 --memory 256Mi
```

> Create a manifest for a VirtualMachineInstancetype with a specified namespace name

Reviewer 2: nit - `namespace`

```shell
virtctl create instancetype --cpu 2 --memory 256Mi --namespace my-namespace
```

Create a manifest for a VirtualMachineInstancetype without a specified namespace name
```shell
virtctl create instancetype --cpu 2 --memory 256Mi --namespaced
```

# Creating Preferences

The virtctl subcommand `create preference` allows easy creation of a preference
manifest from the command line. This command serves as a starting point to create
the basic structure of a manifest, as it does not allow specifying all of the options that
are supported in preferences.

The current set of flags allows us, for example, to specify the preferred CPU topology, machine type
or a storage class.

By default, the command creates the cluster-wide resource. If the user wants to create the namespaced version,
they need to provide the namespaced flag. The namespace name can be specified by using the namespace flag.

For a complete list of flags and their descriptions, use the following command:

```shell
virtctl create preference -h
```

## Examples

Create a manifest for a VirtualMachineClusterPreference with a preferred cpu topology
```shell

> virtctl create preference --cpu-topology preferSockets

Reviewer 2: Idea for a follow up, accept `?` as an argument to `--cpu-topology` to list the available options?

```

> Create a manifest for a VirtualMachinePreference with a specified namespace name

Reviewer 2: nit - As above, drop `name`.

```shell
virtctl create vm --instancetype=my-instancetype --preference=my-preference --volume-pvc=my-pvc
```

> 

Reviewer: Can you move these examples up to the respective subheadings? (Instancetype/Preference). IMO that makes it easier to find them.

Author: Done
